src/gui/editorwidgets/qgsrelationreferencewidget.cpp

```diff
@@ -441,7 +442,7 @@ void QgsRelationReferenceWidget::init()
 
     if ( !mFilterFields.isEmpty() )
     {
-      Q_FOREACH ( const QString &fieldName, mFilterFields )
+      for ( const auto &fieldName : mFilterFields )
```

Comment:
`qgis::as_const`

Reply:
And I'd leave the `QString` instead of `auto`, explicit is better than implicit. Unless the explicit version is overly complicated (like iterators sometimes).